tests/python/test_element_wise.py

```diff
@@ -0,0 +1,118 @@
+import taichi as ti
+from taichi import approx
+from random import random, randint, seed
+import math
+
+
+def _c_mod(a, b):
+    return a - b * int(float(a) / b)
+
+
+def rand(dtype):
+    if ti.core.is_integral(dtype):
+        return randint(1, 5)
+    else:
+        return float(randint(1, 5)) / 5
+
+
+@ti.host_arch_only
+def _test_matrix_element_wise_unary(dtype, ti_func, math_func):
+    n, m = randint(1, 8), randint(1, 8)
+
+    a = ti.Matrix(n, m, dt=ti.f32, shape=())
+    u = ti.Matrix(n, m, dt=ti.f32, shape=())
+
+    w = []
+    for i in range(n * m):
+        w.append(rand(dtype))
+
+    for i in range(n):
+        for j in range(m):
+            u[None][i, j] = w[i * m + j]
+
+    @ti.kernel
+    def func():
+        a[None] = ti_func(u[None])
+
+    func()
+
+    for i in range(n):
+        for j in range(m):
+            expected = math_func(w[i * m + j])
+            assert a[None][i, j] == approx(expected)
+
+
+
+@ti.host_arch_only
+def _test_matrix_element_wise_binary(dtype, ti_func, math_func):
+    n, m = randint(1, 8), randint(1, 8)
```

Comment:
I'm not sure if `5, 7` works indicates `1, 7` works...

Btw, I want to share you a more efficient way to request changes for reviewers:
``````
```suggestion
n, m = randint(1, 8), randint(1, 8)
```
``````
It looks like:
```suggestion
    n, m = 5, 7
```
Then, the PR-author/anyone-with-write-permission can easily click-a-button to commit-the-changes if they feel agree with the suggested changes.

Reply:
> I'm not sure if `5, 7` works indicates `1, 7` works...

Then we should test both cases instead of relying on random number generators. Or at least we leave a comment there saying we are trying to reduce test time.

The feature looks cool! Thanks.

Reply:
> For example, if both randint return 3, then some bugs related to row/col might be hidden.

Since we have enough tests, so it should be nearly-impossible for n and m always equals, so don't worry. Or we may want to test for different n, m combs instead of rand one?